src/BinaryPlatforms.jl

```diff
@@ -692,6 +692,30 @@ function detect_libgfortran_version(libgfortran_name::AbstractString, platform::
     return version
 end
 
+"""
+    probe_dlopen_libgfortran()
+
+Attempts to `dlopen()` libgfortran by guessing its SONAME.
+"""
+function probe_dlopen_libgfortran(;platform::Platform = default_platkey,
+                                   versions::Vector = [3, 4, 5])
+    libgfortran_name(p::MacOS, ver)                = "libgfortran.$(ver).dylib"
+    libgfortran_name(p::Union{Linux,FreeBSD}, ver) = "libgfortran.so.$(ver)"
+    libgfortran_name(p::Windows, ver)              = "libgfortran-$(ver).dll"
+
+    # First, look ONLY in Base.LIBDIR; if that doesn't work, look globally.
+    for prefix in (Base.LIBDIR, "")
```

Comment:
Maybe `Base.LIBDIR` and `joinpath(Base.LIBDIR, "julia")`?  Do we need to prepend `Sys.BINDIR` too?